File: Python/cd/cd.py
import datetime
a = datetime.datetime.now()
while(True):
    inp = input().split()
    N = int(inp[0])
    M = int(inp[1])
    if N==0 and M==0:
        b = datetime.datetime.now()
        print(b-a)
        exit()
    ncd = []
    for i in range(N):
        ncd.append(input())
    mcd = []
    for i in range(M):
        mcd.append(input())
    n = 0
    m = 0
    c = 0
    while(n<N and m<M):
        if ncd[n] < mcd[m]:
            n += 1
        elif ncd[n] > mcd[m]:
            m += 1
        else:
            n += 1
            m += 1
            c +=1
    print(c)

File: Java/cd/cd.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class cd {

     public static void main(String[] args) throws IOException{

        BufferedReader bi = new BufferedReader(new InputStreamReader(System.in));
        int N, M;
        String[] inp;

        while (true){
            String line = "";
                line = bi.readLine();
            inp = line.split(" ");
            N = Integer.parseInt(inp[0]);
            M = Integer.parseInt(inp[1]);
            if (N == 0 && M == 0) break;
            int[] ncd = new int[N];
            for (int i = 0; i < N; i++){
                    ncd[i] = Integer.parseInt(bi.readLine());
            }
            int[] mcd = new int[M];
            for (int i = 0; i < M; i++){
                    mcd[i] = Integer.parseInt(bi.readLine());
            }

            int n = 0;
            int m = 0;
            int c = 0;
            while (n<N && m<M){
                if (ncd[n] < mcd[m]) n++;
                else if (ncd[n] > mcd[m]) m++;
                else {
                    n++;
                    m++;
                    c++;
                }
            }
            System.out.println(c);
        }
    }
}
